src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -37,35 +39,207 @@ static InitChainEntry D_8098EE8C[] = {
-extern UNK_TYPE D_06000B70;
-extern UNK_TYPE D_06001040;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E5C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E62C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E8A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E900.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E9C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/ObjKibako2_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/ObjKibako2_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098EB78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098EC68.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098ED20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/ObjKibako2_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/ObjKibako2_Draw.s")
+extern Gfx D_06000960[];
+extern CollisionHeader D_06000B70;
+extern Gfx D_06001040[];
+
+s32 ObjKibako2_ContainsSkulltula(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 actorSpawnParam;
+    s32 flag;
+
+    actorSpawnParam = KIBAKO2_SKULLTULA_SPAWN_PARAM(this);
+    flag = -1;
+    if ((u16) actorSpawnParam & 3) {
+        flag = ((actorSpawnParam & 0x3FC) >> 2) & 0xFF;
+    }
+    return !((flag >= 0) && Actor_GetChestFlag(globalCtx, flag));
+}
+
+void ObjKibako2_Break(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Vec3f* thisPos;
+    Vec3f pos;
+    Vec3f velocity;
+    s16 angle;
+    s32 i;
+
+    thisPos = &this->dyna.actor.world.pos;
+    for (i = 0, angle = 0; i < 0x10; i++, angle += 0x4E20) {
+        f32 sn = Math_SinS(angle);
+        f32 cs = Math_CosS(angle);
+        f32 tempRand;
+        s32 phi_s0;
+
+        tempRand = Rand_ZeroOne() * 30.0f;
+        pos.x = sn * tempRand;
+        pos.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        pos.z = cs * tempRand;
+        velocity.x = pos.x * 0.2f;
+        velocity.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        velocity.z = pos.z * 0.2f;
+        pos.x += thisPos->x;
+        pos.y += thisPos->y;
+        pos.z += thisPos->z;
+        tempRand = Rand_ZeroOne();
+        if (tempRand < 0.05f) {
+            phi_s0 = 0x60;
+        } else if (tempRand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -200, phi_s0, 28, 2, 0, (Rand_ZeroOne() * 30.0f) + 5.0f,
+                            0, 0, 70, KAKERA_COLOR_NONE, OBJECT_KIBAKO2, D_06001040);
+    }
+    func_800BBFB0(globalCtx, thisPos, 90.0f, 6, 100, 160, 1);
+}
+
+
+void ObjKibako2_SpawnCollectible(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 collectible;
+
+    collectible = func_800A8150(this->dyna.actor.params & 0x3F);
```

Comment:
Usually compound params need macros and an enum, but in this case we might be better off waiting for all the "container" actors to be done.

Reply:
I would prefer to leave this as-is, both because as you said, "container" actors are not done, but also to have consistency with Kibako.

Reply:
Actually, comparing it to OoT and looking at how `func_800A81A4` uses the parameters passed into it, it's pretty easy to figure out what this means. So I rescind that previous comment. I'll put a macro for this too, and when this gets approved, I'll do a PR for Kibako to ensure consistency with this.